tests/fib/test_fib.py

```diff
@@ -67,41 +76,55 @@ def get_fib_info(duthost, cfg_facts, mg_facts):
             }
     """
     timestamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
-    duthost.shell("redis-dump -d 0 -k 'ROUTE*' -y > /tmp/fib.{}.txt".format(timestamp))
-    duthost.fetch(src="/tmp/fib.{}.txt".format(timestamp), dest="/tmp/fib")
-
-    po = cfg_facts.get('PORTCHANNEL', {})
-    ports = cfg_facts.get('PORT', {})
-
     fib_info = {}
-    with open("/tmp/fib/{}/tmp/fib.{}.txt".format(duthost.hostname, timestamp)) as fp:
-        fib = json.load(fp)
-        for k, v in fib.items():
-            skip = False
-
-            prefix = k.split(':', 1)[1]
-            ifnames = v['value']['ifname'].split(',')
-            nh = v['value']['nexthop']
-
-            oports = []
-            for ifname in ifnames:
-                if po.has_key(ifname):
-                    oports.append([str(mg_facts['minigraph_ptf_indices'][x]) for x in po[ifname]['members']])
-                else:
-                    if ports.has_key(ifname):
-                        oports.append([str(mg_facts['minigraph_ptf_indices'][ifname])])
+    for asic_index, asic_cfg_facts in  enumerate(cfg_facts):
+
+        asic = duthost.asic_instance(asic_index)
+
+        asic.shell("{} redis-dump -d 0 -k 'ROUTE*' -y > /tmp/fib.{}.txt".format(asic._ns_arg, timestamp))
```

Comment:
why not use "asic.command()". No need to use "asic._ns_arg" which looks like a private attribute as per convention.

Reply:
@smaheshm  the ansible command module doesnt work with `>` 
https://github.com/ansible/ansible/blob/f9f839fa08eee46ad7a86d6cbc7519541a50c7ef/lib/ansible/modules/command.py#L22. `shell` module is recommended for such cases

Change the `ns_arg` to public variable